### ISSUE TITLE
Handle non-printable characters

### ISSUE THREAD

**Issue by shaag7967** [shaag7967/multiserialviewer#9]
Right now, only printable characters are shown in the SerialViewer window. If non-printable characters are received, they get discarded.

Normally non-ascii characters are not received (in a normal debugging environment). But it could be interesting to see, that data is received (even if not printable) e.g. if the baudrate is not configured correctly (either on the device side or the tool).

If the non printable bytes are printed as hex, then you will see them probably when doing a power cycle (of the device). But this should not be a major problem.

Another option would be to have a kind of separate log (not sure if this is worth the effort).

**Comment by shaag7967** [comment]
@Quantum1337: And/or we add a counter to the statistics feature and show there how many non-ascii characters we received?

**Comment by Quantum1337** [comment]
yes that is reasonable. What about decoding utf-8 instead of ascii then? utfi-8 is downwards compatible to ascii and a de-facto standard. It can also encode German umlauts as äüö, and other special characters from other languages. 
I know its not that common on embedded systems. But I don't see a downside currently except the algorithem wich is slightly more complex when it detects a non ascii character.

**Comment by shaag7967** [comment]
yes, sure. Good idea

**Comment by shaag7967** [comment]
I think the problem with utf8 is, that we are processing a byte stream and get the data byte by byte. So in many cases we do not have a complete/valid utf8 character (correct me if I'm wrong). To solve this we would need to detect if a complete char was received, if not, cache it and wait for the following bytes (this needs a timeout and so on...).

For now, I defined a set of accepted bytes and if we receive an "invalid" byte, I convert it into its hex representation. This looks like this e.g.:
```
asfsdfasf
zzz777[08][88][00][8A]qwerqwer
```
I also added a highlighting pattern for the hex bytes.

See Commit 5dbd137 #

**Comment by shaag7967** [comment]
-> for now, only bytes "b == 0x0D or b == 0x0A or 32 <= b <= 126" are printed. Everything else is shown as hex (if enabled). Lets handle utf8 in a later version.
